include/boost/next_prior.hpp

```diff
@@ -140,6 +138,24 @@ struct prior_impl1< T, Distance, true >
     }
 };
 
+template< typename T, typename Distance, bool IsIterator = is_iterator< T >::value >
+struct prior_advance_impl :
+    public prior_minus_assign_impl< T, Distance >
+{
+};
+
+template< typename T, typename Distance >
+struct prior_advance_impl< T, Distance, true >
+{
+    static T call(T x, Distance n)
+    {
+        // Avoid negating n to sidestep possible integer overflow
+        std::reverse_iterator< T > rx(x);
```

Comment:
> Whether the operator requires `difference_type` or allows something else depends on the iterator.

OK, understood.

> Unfortunately, `boost::advance` uses `difference_type` for the distance argument

I implemented it in that way, because I followed the way of `std::next/prev` and `std2::advance`. I prefer the current implementation, but I'm not stuck to it; you can submit a PR to change the implementation. 
(Disclaimer: I'm not a maintainer of Boost.Iterator.)